web-local/src/lib/svelte-query/actions.ts

```diff
@@ -0,0 +1,83 @@
+import { goto } from "$app/navigation";
+import {
+  getRuntimeServiceListFilesQueryKey,
+  V1DeleteFileAndReconcileResponse,
+  V1RenameFileAndReconcileResponse,
+} from "@rilldata/web-common/runtime-client";
+import type { ActiveEntity } from "@rilldata/web-local/common/data-modeler-state-service/entity-state-service/ApplicationEntityService";
+import type { EntityType } from "@rilldata/web-local/common/data-modeler-state-service/entity-state-service/EntityStateService";
+import { getNextEntityName } from "@rilldata/web-local/common/utils/getNextEntityId";
+import { dataModelerService } from "@rilldata/web-local/lib/application-state-stores/application-store";
+import { commonEntitiesStore } from "@rilldata/web-local/lib/application-state-stores/common-store";
+import {
+  getFileFromName,
+  getLabel,
+  getRouteFromName,
+} from "@rilldata/web-local/lib/components/entity-mappers/mappers";
+import notifications from "@rilldata/web-local/lib/components/notifications";
+import { queryClient } from "@rilldata/web-local/lib/svelte-query/globalQueryClient";
+import type { UseMutationResult } from "@sveltestack/svelte-query";
+
+export async function renameEntity(
```

Comment:
Let's use `renameFile` and `deleteFile` to mirror the terminology in the backend? I'd prefer we avoid the term "Entity", which isn't specific.